comm/src/main/scala/coop/rchain/p2p/effects/PacketHandler.scala

```diff
@@ -13,9 +13,7 @@ object PacketHandler extends PacketHandlerInstances {
 
   def forTrans[F[_]: Monad, T[_[_], _]: MonadTrans](
       implicit C: PacketHandler[F]): PacketHandler[T[F, ?]] =
-    new PacketHandler[T[F, ?]] {
-      def handlePacket(packet: Packet): T[F, String] = C.handlePacket(packet).liftM[T]
-    }
+    (packet: Packet) => C.handlePacket(packet).liftM[T]
```

Comment:
Lambda to class conversion? I was not aware...

Reply:
This is called SAM conversion, see http://www.scala-lang.org/news/2.12.0/#lambda-syntax-for-sam-types